apps/zotonic_mod_base/src/controllers/controller_file.erl

```diff
@@ -142,11 +142,30 @@ process(_Method, _AcceptedCT, _ProvidedCT, Context) ->
             MaxAge = z_context:get(max_age, Context1, ?MAX_AGE),
             Context2 = set_cache_control_public(is_public(Info#z_file_info.acls, Context1), MaxAge, Context1),
             Context3 = set_content_policy(Info, Context2),
-            {z_file_request:content_stream(Info, cowmachine_req:resp_content_encoding(Context3)), Context3}
+            Context4 = z_context:set_resource_headers(maybe_id(Context3), Context3),
+            {z_file_request:content_stream(Info, cowmachine_req:resp_content_encoding(Context4)), Context4}
     end.
 
 %%%%% -------------------------- Support functions ------------------------
 
+maybe_id(Context) ->
+    case z_context:get(?MODULE, Context) of
+        {error, _} ->
+            {false, Context};
```

Comment:
Erlang does not really have an "option/maybe" type.. But I think this function should either return an id or undefined. It looks like the spec from `set_resource_headers` also expects that.

Reply:
Good one.  That should have been `false` without Context.